core/src/main/scala/tech/beshu/ror/utils/SSLCertHelper.scala

```diff
@@ -0,0 +1,195 @@
+    sslConfiguration.truststoreFile
+      .map { truststoreFileName =>
+        Resource
+          .fromAutoCloseable(IO(new FileInputStream(truststoreFileName.value)))
+          .use { truststoreFile =>
+            IO {
+              val keystore = if (fipsCompliant) {
+                logger.info("Trying to load truststore in FIPS compliant BCFKS format...")
+                java.security.KeyStore.getInstance("BCFKS", "BCFIPS")
+              } else {
+                logger.info("Trying to load truststore in JKS or PKCS#12 format...")
+                java.security.KeyStore.getInstance("JKS")
+              }
+              keystore.load(truststoreFile, sslConfiguration.truststorePassword)
+              keystore
+            }
+          }
+      }.sequence
+  }
+
+  private def prepareAlias(keystore: KeyStore, config: SslConfiguration) =
+    config.keyAlias match {
+      case None if keystore.aliases().hasMoreElements =>
+        val firstAlias = keystore.aliases().nextElement()
+        logger.info(s"ROR SSL: ssl.key_alias not configured, took first alias in keystore: $firstAlias")
+        firstAlias
+      case None =>
+        throw MalformedSslSettings("No alias found, therefore key found in keystore!")
+      case Some(keyAlias) =>
+        keyAlias.value
+    }
+
+  private def removeAllAliasesFromKeystoreBesidesOne(keystore: KeyStore, alias: String): Unit = {
+    val unnecessaryAliases = keystore.aliases().asScala.toSet.-(alias)
+    unnecessaryAliases.foreach(keystore.deleteEntry)
+  }
+
+  def getTrustManagerFactory(sslConfiguration: SslConfiguration, fipsCompliant: Boolean): TrustManagerFactory = {
+    Try(loadTruststore(sslConfiguration, fipsCompliant)
+      .map {
+        _.map { truststore =>
+          val tmf = getTrustManagerFactoryInstance(fipsCompliant)
+          tmf.init(truststore)
+          tmf
+        }
+      }
+      .unsafeRunSync()) match {
+      case Success(kmf) => kmf.getOrElse(throw TrustManagerNotConfiguredException)
+      case Failure(exception) =>
+        throw UnableToInitializeTrustManagerFactoryUsingProvidedTruststore(exception.getMessage)
+    }
+  }
+
+  private def getKeyManagerFactory(sslConfiguration: SslConfiguration, fipsCompliant: Boolean): IO[KeyManagerFactory] = {
+    loadKeystore(sslConfiguration, fipsCompliant)
+      .map { keystore =>
+        removeAllAliasesFromKeystoreBesidesOne(keystore, prepareAlias(keystore, sslConfiguration))
+        val kmf = getKeyManagerFactoryInstance(fipsCompliant)
+        kmf.init(keystore, sslConfiguration.keystorePassword)
+        kmf
+      }
+  }
+
+  def prepareSSLContext(sslConfiguration: SslConfiguration, fipsCompliant: Boolean): SslContext = {
+    Try(SSLCertHelper
+      .getKeyManagerFactory(sslConfiguration, fipsCompliant)
+      .unsafeRunSync()) match {
+      case Success(keyManagerFactory) =>
+        val sslCtxBuilder = SslContextBuilder.forServer(keyManagerFactory)
```

Comment:
what about moving it to the map method and calling unsafeRunSync at the end of monadic chain?